specs/CookieManagement.md

```diff
@@ -0,0 +1,491 @@
+        /// equivalent cookies if they exist.
+        void AddOrUpdateCookie(CoreWebView2Cookie cookie);
+
+        /// Deletes a cookie whose params matches those of the specified cookie.
+        void DeleteCookie(CoreWebView2Cookie cookie);
+
+        /// Deletes browser cookies with matching name and uri or domain/path pair.
+        /// Cookie name is required.
+        /// If uri is specified, deletes all cookies with the given name where domain
+        /// and path match provided URI.
+        /// If domain is specified, deletes only cookies with the exact domain.
+        /// If path is specified, deletes only cookies with the exact path.
+        void DeleteCookies(String name, String uri, String Domain, String Path);
+
+        /// Deletes all cookies under the same profile.
+        /// This could affect other WebViews under the same user profile.
+        void DeleteAllCookies();
+    }
+
+    /// A list of cookie objects.
+    runtimeclass CoreWebView2CookieList : Windows.Foundation.Collections.IVector<CoreWebView2Cookie>
+    {
+        /// The number of cookies contained in the CoreWebView2CookieList.
+        uint Size { get; };
+
+        /// Get the cookie object at the given index.
+        CoreWebView2Cookie GetAt(uint index);
+    }
+
+    /// Provides a set of properties that are used to manage a CoreWebView2Cookie.
+    runtimeclass CoreWebView2Cookie
+    {
+        /// Cookie name.
+        String Name { get; };
+
+        /// Cookie value.
+        String Value { get; set; };
+
+        /// The domain for which the cookie is valid.
+        /// The default is the host that this cookie has been received from.
+        String Domain { get; };
+
+        /// The path for which the cookie is valid. If not specified, this cookie
+        /// will be sent to all pages on the Domain.
+        String Path { get; };
+
+        /// The expiration date and time for the cookie as the number of seconds since the UNIX epoch.
+        /// The default is -1.0, which means cookies are session cookies by default.
+        Double Expires { get; set; };
+
+        /// Whether this cookie is http-only.
+        /// True if a page script or other active content cannot access this
+        /// cookie. The default is false.
+        Boolean IsHttpOnly { get; set; };
+
+        /// SameSite status of the cookie which represents the enforcement mode of the cookie.
+        /// The default is CoreWebView2CookieSameSiteKind.Lax.
+        CoreWebView2CookieSameSiteKind SameSite { get; set; };
+
+        /// The security level of this cookie. True if the client is only to return
+        /// the cookie in subsequent requests if those requests use HTTPS.
+        /// The default is false.
+        /// Note that cookie that requests CoreWebView2CookieSameSiteKind.None but
+        /// is not marked Secure will be rejected.
+        Boolean IsSecure { get; set; };
+
+        /// Converts a System.Net.Cookie to a CoreWebView2Cookie.

```

Comment:
Maybe just add a comment that these two are only for the .NET API and not the WinRT API